Mathlib/MeasureTheory/Integral/Periodic.lean

```diff
@@ -283,8 +328,8 @@ section RealValued
 
 open Filter
 
-variable {g : ℝ → ℝ}
-variable (hg : Periodic g T) (h_int : ∀ t₁ t₂, IntervalIntegrable g MeasureSpace.volume t₁ t₂)
+variable {g : ℝ → ℝ} {t₁ : ℝ}
+variable (hg : Periodic g T) (h_int : IntervalIntegrable g MeasureSpace.volume t₁ (t₁ + T))
```

Comment:
I'm not sure I see the point in having a `t_1` here: all these assumptions are equivalent, so why not go for the one which is easiest to state, i.e., `IntervalIntegrable g MeasureSpace.volume 0 T`? Maybe it's worth having a specialized version of your result above for this common specific case, by the way (the other one does not apply directly as `T` and `0 + T` are not the same thing)

Reply:
Added a specialized version, as you suggested. Thanks for the suggestion.

Reply:
Still, in this whole paragraph, I think you should replace `h_int` with `(h_int : IntervalIntegrable g MeasureSpace.volume 0 T)`. This is the simplest form of several equivalent conditions, so we should probably take this one as the standard formulation.

Reply:
@sgouezel Implemented your suggestion, except for the theorems `intervalIntegral_add_eq_add` and `intervalIntegral_add_zsmul_eq`, which are different in that they have no positivity assumption of the length of the period.